pallets/elections/src/lib.rs

```diff
@@ -0,0 +1,163 @@
+#![cfg_attr(not(feature = "std"), no_std)]
+
+#[cfg(test)]
+mod mock;
+#[cfg(test)]
+mod tests;
+
+use frame_support::traits::StorageVersion;
+pub use pallet::*;
+
+/// The current storage version.
+const STORAGE_VERSION: StorageVersion = StorageVersion::new(0);
+
+#[frame_support::pallet]
+pub mod pallet {
+    use super::*;
+    use frame_election_provider_support::{
+        ElectionDataProvider, ElectionProvider, Support, Supports,
+    };
+    use frame_support::pallet_prelude::*;
+    use frame_system::{
+        ensure_root,
+        pallet_prelude::{BlockNumberFor, OriginFor},
+    };
+    use primitives::{
+        SessionIndex, DEFAULT_MILLISECS_PER_BLOCK, DEFAULT_SESSIONS_PER_ERA, DEFAULT_SESSION_PERIOD,
+    };
+    use sp_std::{collections::btree_map::BTreeMap, prelude::Vec};
+
+    #[pallet::storage]
+    #[pallet::getter(fn members)]
+    pub type Members<T: Config> = StorageValue<_, Vec<T::AccountId>, ValueQuery>;
+
+    #[pallet::config]
+    pub trait Config: frame_system::Config {
+        type Event: From<Event<Self>> + IsType<<Self as frame_system::Config>::Event>;
+        type DataProvider: ElectionDataProvider<Self::AccountId, Self::BlockNumber>;
+    }
+
+    #[pallet::event]
+    #[pallet::generate_deposit(pub(super) fn deposit_event)]
+    pub enum Event<T: Config> {
+        ChangeMembers(Vec<T::AccountId>),
```

Comment:
Also, a concrete and important case study: that happens if I send this extrinsic in the last session of an era?

Reply:
In the previous version, pallet-aleph returned a list of authorities directly in the implementation of SessionManager::new_session. Here pallet staking implements this method and inside it it calls elect instead of reading directly.
The only difference is that we previously had also a way of delaying elections with `session_for_validators_change`. Now elections are for an era `n` which starts in session `k` happen in `on_initialize` hook of last block of session `k-2`.

I suppose that it's now obvious, that in your example the change triggered by the extrinsic will take place in the era after the era that comes after this last session you send the extrinsic.

Reply:
Ok, so definitely a comment is required that explain exactly what will be the effect of sending this extrinsic. 

But also I think this sentence "The only difference is that ..." is not quite correct, because we move from the granularity of sessions to eras. This is a huge difference that gives us way less freedom when reacting to emergencies.

Reply:
We can sent this xt and the [force_new_era](https://paritytech.github.io/substrate/master/pallet_staking/pallet/enum.Call.html#variant.force_new_era) to have the same functionality.

Reply:
Nice, I think documenting all that in a comment would make a lot of sense.

Reply:
Maybe a notion doc would be a better place to describe it?

Reply:
Hmm, yes, it would certainly be more convenient to read. But then it would still be good to somehow indicate in the source where is this documented... (Not sure a link would make sense though, as github is public and notion is private...)

Reply:
I don't see why it should be noted here. Even stronger, for me such comments don't fit here. I would document how to achieve some goals by combination of various call in our docs not in here.

Reply:
Fine, makes sense. The only thing I would like to make sure is that we have a precise documentation what this call does, so that in case of emergency we can look it up.

Reply:
Added a notion doc.